File: solar-panel-installation-system/src/Service/ProjectService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/project/";


//Összes project lekérése
const findAll = () => {

    return axios.get(API_URL + "findall");
}

//Új projekt létrehozása
const createProject = (project) => {

    return axios.post(API_URL + "createproject", project)
}

//Meglévő projekt módosítása
//Alkatrész módosítása
const modifyProject = (id, project) => {

    return axios.put(API_URL + "update/" + id, project);

};

//Keresés id szerint
const findById = (id) => {

    return axios.get(API_URL + "findbyid/" + id);

};

//Alkatrész lefoglalása
const reservePart = (projectId, partId, reservedNumber) => {

    return axios.get(API_URL + "reservepart/" + projectId + "/" + partId + "/" + reservedNumber)
}

//Előfoglalás leadása az alkatrészre
const preReservePart = (projectId, partId, preReservedNumber) => {
    return axios.get(API_URL + "prereservepart/" + projectId + "/" + partId + "/" + preReservedNumber)

}

//Árkalkuláció elkészítése. Ha minden alkatrész elérhető a raktárban, visszatér a költséggel és a project "scheduled" fázisba kerül.
//Ha nem érhető el minden alkatrész, akkor a költség 0 és a project "wait" fázisba kerül
const showFullCost = (projectId) => {
    return axios.get(API_URL + "showfullcost/" + projectId)
}

//Projekt lezárása.
//Ha sikeres akkor "completed" fázsiba kerül. Ha nem, akkor "failed" fázisba.
const finishProject = (projectId, status) => {
    return axios.get(API_URL + "finishproject/" + projectId + "/" + status)
}

//Projektek listázása kivételezésre
const listProjectsWithoutPreReservation = () => {
    return axios.get(API_URL + "listprojectswithoutprereservation")
}

//A kiválasztott projekthez tartozó lefoglalt alkatrészek listázása
const showPartsOfProject = (projectId) => {
    return axios.get(API_URL + "showpartsofproject/" + projectId)
}

//Legjobb útvonal listázása
const bestPath = (projectId) => {
    return axios.get(API_URL + "bestpath/" + projectId)
}


//Projektek lekérése
//Folyamatban lévő projektek vagy lezárt projektek
const findByProjectStatus = (status) => {
    return axios.get(API_URL + "findbyprojectstatus/" + status)

}

const ProjectService = {
    findAll,
    createProject,
    modifyProject,
    findById,
    reservePart,
    preReservePart,
    showFullCost,
    finishProject,
    listProjectsWithoutPreReservation,
    showPartsOfProject,
    bestPath,
    findByProjectStatus

};

export default ProjectService;

File: Solar-panel-springboot/src/main/java/com/jwt_authentication_springboot/service/ProjectService.java
package com.jwt_authentication_springboot.service;

import com.jwt_authentication_springboot.model.Box;
import com.jwt_authentication_springboot.model.Part;
import com.jwt_authentication_springboot.model.Project;
import com.jwt_authentication_springboot.payload.response.BestPathDTO;
import com.jwt_authentication_springboot.payload.response.PartDTO;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface ProjectService {

    public void save(Project project);

    public ResponseEntity<Project> findById(long id);

    List<Project> findAll();

    void update(Long id, Project project);

    void reservePart(Long projectId, Long partId, int reservedNumber);

    void preReservePart(Long projectId, Long partId, int preReservedNumber);

    //Árkalkuláció elkészítése. Ha minden alkatrész elérhető a raktárban, visszatér a költséggel és a project "scheduled" fázisba kerül.
    //Ha nem érhető el minden alkatrész, akkor a költség 0 és a project "wait" fázisba kerül
    int showFullCost(Long projectId);

    //Projekt lezárása
    //A bejövő paraméter "success" vagy "unsuccess"
    //Ha sikeres akkor "completed" fázsiba kerül, ha nem, akkor "failed" fázisba.
    void finishProject(Long projectId, String status);

    //Projektek listázása kivételezésre
    //Csak olyan projekteket adok vissza, amelyeknek a kivételezését meg lehet kezdeni
    //Tehát az adott projekthez nem tartozik előfoglalt alkatrész
    List<Project> listProjectsWithoutPreReservation();

    //A kiválasztott projekthez tartozó lefoglalt alkatrészek listázása
    //Ezeket az alkatrészeket már ki lehet venni a raktárból
    List<PartDTO> showPartsOfProject(Long projectId);

    //Legjobb útvonal megkeresése az alkatrészek összegyűjtéséhez
    List<BestPathDTO> bestPath(Long projectId);

    //Projektek lekérése
    //Folyamatban lévő projektek vagy lezárt projektek
    List<Project> findByProjectStatus(String status);
}
